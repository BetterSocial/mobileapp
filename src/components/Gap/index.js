import * as React from 'react';
import {View} from 'react-native';

const Gap = ({width, height, style}) => {
  return <View style={[style, {width: width, height: height}]} />;
};

export default Gap;
