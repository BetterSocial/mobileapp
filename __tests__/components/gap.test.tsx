import React from 'react';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Gap from '../../src/components/Gap';

configure({adapter: new Adapter()});

describe('component gap', () => {
  it('gap snapshot', () => {
    const component = shallow(<Gap style={{width: 20}} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('gap renders correctly', () => {
    const tree = renderer.create(<Gap style={{width: 20}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
