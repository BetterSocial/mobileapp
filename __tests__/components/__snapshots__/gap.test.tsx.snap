// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component gap gap renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "width": 20,
      },
      Object {
        "height": undefined,
        "width": undefined,
      },
    ]
  }
/>
`;

exports[`component gap gap snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "width": 20,
      },
      Object {
        "height": undefined,
        "width": undefined,
      },
    ]
  }
/>
`;
